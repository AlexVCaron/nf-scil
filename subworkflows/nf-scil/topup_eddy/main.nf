// ** Importing modules from nf-scil ** //
include { TOPUP } from '../../../modules/nf-scil/preproc/topup/main'
include { EDDY } from '../../../modules/nf-scil/preproc/eddy/main'

workflow TOPUP_EDDY {

    // **
    // **
    // **

    take:
        ch_image // channel: [ val(meta), [ dwi, bval, bvec, b0, rev_dwi, rev_bval, rev_bvec, rev_b0 ] ]
        config_topup // channel: [ config_topup ]

    main:

        ch_versions = Channel.empty()

        if ( config_topup and ( ch_image.rev_dwi or ch_image.rev_b0 ))
        {
            TOPUP ( ch_image, config_topup )
            ch_versions = ch_versions.mix(TOPUP.out.versions.first())
            EDDY ( [ch_image.dwi, ch_image.bval, ch_image.bvec, ch_image.b0, ch_image.rev_dwi, ch_image.rev_bval, ch_image.rev_bvec, ch_image.rev_b0, TOPUP.out.topup_corrected_b0s, TOPUP.out.topup_fieldcoef, TOPUP.out.topup_movpart] )
        }
        else
        {
            EDDY ( ch_image )
        }

        ch_versions = ch_versions.mix(EDDY.out.versions.first())

    emit:
        dwi      = EDDY.out.dwi_corrected       // channel: [ val(meta), [ dwi_corrected ] ]
        bval     = EDDY.out.bval_corrected      // channel: [ val(meta), [ bval_corrected ] ]
        bvec     = EDDY.out.bvec_corrected      // channel: [ val(meta), [ bvec_corrected ] ]
        b0_mask  = EDDY.out.b0_mask             // channel: [ val(meta), [ b0_mask ] ]
        versions = ch_versions                  // channel: [ versions.yml ]
}
