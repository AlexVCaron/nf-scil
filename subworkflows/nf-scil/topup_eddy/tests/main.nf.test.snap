{
    "topup_eddy": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_corrected.nii.gz:md5,09d850a2b7460ab6a6159d14a93a02d7"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bval_eddy:md5,be4e72882f347bec4fcf5380f900719d"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_eddy_corrected.bvec:md5,97e948b33375382034e0cc2ba4a437c2"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_b0.nii.gz:md5,9abd674ae7018880c17e9aa2b7d9864d"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_bet_mask.nii.gz:md5,fc1f94e2a5e2cf5197d7fdc28a83ca28"
                    ]
                ],
                "5": [
                    "versions.yml:md5,5af3f2becfc5b99563365a86fc38fc21",
                    "versions.yml:md5,76cab382c75fc0f861e01b34532789a4",
                    "versions.yml:md5,f4dfd5f1c68f28be5dc9abe67eb673e4"
                ],
                "b0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_b0.nii.gz:md5,9abd674ae7018880c17e9aa2b7d9864d"
                    ]
                ],
                "b0_mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_bet_mask.nii.gz:md5,fc1f94e2a5e2cf5197d7fdc28a83ca28"
                    ]
                ],
                "bval": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bval_eddy:md5,be4e72882f347bec4fcf5380f900719d"
                    ]
                ],
                "bvec": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_eddy_corrected.bvec:md5,97e948b33375382034e0cc2ba4a437c2"
                    ]
                ],
                "dwi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_corrected.nii.gz:md5,09d850a2b7460ab6a6159d14a93a02d7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5af3f2becfc5b99563365a86fc38fc21",
                    "versions.yml:md5,76cab382c75fc0f861e01b34532789a4",
                    "versions.yml:md5,f4dfd5f1c68f28be5dc9abe67eb673e4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T18:13:20.489862"
    }
}